### PR TITLE
updating BuildConfig retry

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

**Comment by rcerven** [comment]
working on test

**Comment by twaugh** [comment]
Can't `retry_on_conflict` help somehow? Seems a shame to duplicate logic.

File: osbs/utils.py [hunk 1]
```diff
@@ -554,3 +556,22 @@ def get_name_and_value(self, label_type):
...
+def retry_on_conflict(func,
+                      retry_times=OS_CONFLICT_MAX_RETRIES,
+                      retry_delay=OS_CONFLICT_WAIT):
+    @wraps(func)
+    def retry(*args, **kwargs):
+        for counter in range(retry_times + 1):
+            try:
+                return func(*args, **kwargs)
+            except OsbsResponseException as ex:
+                if ex.status_code == 409 and counter != retry_times:
```
Comment (vrutkovs) [inline osbs/utils.py]: Use `http_client.CONFLICT` instead of `409`

Comment (rcerven) [reply osbs/utils.py]: ok

File: osbs/utils.py [hunk 2]
```diff
@@ -554,3 +557,22 @@ def get_name_and_value(self, label_type):
...
             return (label_type, self._df_labels[label_type])
+
+
+def retry_on_conflict(func,
+                      retry_times=OS_CONFLICT_MAX_RETRIES,
+                      retry_delay=OS_CONFLICT_WAIT):
+    @wraps(func)
+    def retry(*args, **kwargs):
+        for counter in range(retry_times + 1):
+            try:
```
Comment (vrutkovs) [inline osbs/utils.py]: We should log retries, that would help with debugging

File: tests/test_core.py [hunk 3]
```diff
@@ -265,7 +266,7 @@ def test_get_multiple_build_config_by_labels(self, openshift):  # noqa
         ([http_client.CONFLICT, http_client.OK], False),
         ([http_client.CONFLICT, http_client.CONFLICT, http_client.UNAUTHORIZED], True),
         ([http_client.UNAUTHORIZED], True),
-        ([http_client.CONFLICT for _ in range(10)], True),
+        ([http_client.CONFLICT for _ in range(OS_CONFLICT_MAX_RETRIES + 1)], True),
```
Comment (vrutkovs) [inline tests/test_core.py]: Is there a test case where it fails after all attempts are depleted?

Comment (rcerven) [reply tests/test_core.py]: that test is in test_utils

and I'll make 1 more test which will conflict and success after

**Pull request merged**